include/sys/zfs_sa.h

```diff
@@ -87,6 +88,8 @@ typedef enum zpl_attr {
 #define	SA_UID_OFFSET		24
 #define	SA_GID_OFFSET		32
 #define	SA_PARENT_OFFSET	40
+#define	SA_FLAGS_OFFSET		48
+#define	SA_PROJID_OFFSET	128
```

Comment:
I'm still not entirely clear on this.  I see that in the `zfs_sa_upgrade()` case PROJID will be placed after LINKS since `sa_replace_all_by_template()` preserves the passed order.  However, I don't see where this ordering is made the preferred default layout for new filesystems.  Can you point me to the code which ensures this is true.